### PR TITLE
test: fix manifest e2e test assumption

### PR DESCRIPTION
**Pull request by giuseppe**
fix an assumption in the test, that the --compression-format is unchanged from the original image.

Instead validate that all the required architectures are part of the manifest.

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: test/e2e/manifest_test.go [hunk 1]
```diff
@@ -16,6 +17,50 @@ import (
...
 
+// validateManifestHasAllArchs checks that the specified manifest has all
+// the archs in `imageListInstance`
+func validateManifestHasAllArchs(path string) error {
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return err
+	}
+
+	var result map[string]interface{}
```
Comment (mtrmac) [inline test/e2e/manifest_test.go:28]: Absolutely non-blocking: Defining a function-private type might be shorter and easier to read:
```go
var result struct {
   Manifests []struct {
      Platform struct {
         Architecture string
      }
   }
}
```
or something like that.

OTOH this works just fine, and passes tests as is.

**Review by mtrmac — COMMENTED** [review]
Thanks!

Just one comment nit, really.

File: test/e2e/manifest_test.go (hunk 1 above)
Comment (giuseppe) [reply test/e2e/manifest_test.go]: your suggestion is better, fixed now

**Comment by TomSweeneyRedHat** [comment]
LGTM

**Comment by mtrmac** [comment]
Thanks again!

**Pull request merged**